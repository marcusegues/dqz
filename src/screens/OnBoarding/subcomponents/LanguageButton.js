// @flow
import React from 'react';
import Touchable from 'react-native-platform-touchable';
import { Text } from 'react-native';
import { languageButton, languageButtonText } from '../styles/languageButton';
import type {
  LanguageButtonSizeType,
  LanguageButtonSelectedType,
  LanguageCodeType,
} from '../types';

type PropsType = {
  size: LanguageButtonSizeType,
  selected: LanguageButtonSelectedType,
  languageCode: LanguageCodeType,
  onPress: () => any,
};

export const LanguageButton = (props: PropsType) => {
  const { size, selected, languageCode, onPress } = props;
  return (
    <Touchable
      key={languageCode}
      background={Touchable.Ripple('#006699')}
      style={languageButton({
        size,
        selected,
      })}
      onPress={onPress}
    >
      <Text
        style={languageButtonText({
          size,
          selected,
        })}
      >
        {languageCode.toUpperCase()}
      </Text>
    </Touchable>
  );
};
