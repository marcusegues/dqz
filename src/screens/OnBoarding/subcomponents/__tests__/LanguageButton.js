import renderer from 'react-test-renderer';
import React from 'react';
import { LanguageButton } from '../LanguageButton';

describe('LanguageButton', () => {
  test('renders according to snapshot', () => {
    const component = renderer
      .create(
        <LanguageButton text="text" onPress={jest.fn()} languageCode="de" />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  test('renders according to snapshot', () => {
    const component = renderer
      .create(
        <LanguageButton
          text="text"
          onPress={jest.fn()}
          languageCode="de"
          selected
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
