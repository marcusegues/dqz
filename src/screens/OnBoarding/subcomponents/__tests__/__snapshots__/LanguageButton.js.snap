// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageButton "de" language and "size=small" renders according to snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#fff",
      "borderBottomWidth": 5,
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "paddingHorizontal": 55.714285714285715,
      "paddingVertical": 21.57941176470588,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 1.5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "roboto_regular",
        "fontSize": 28.285714285714285,
        "textAlign": "center",
      }
    }
  >
    DE
  </Text>
</View>
`;

exports[`LanguageButton "en" language and "size=small" renders according to snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#fff",
      "borderBottomWidth": 5,
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "paddingHorizontal": 55.714285714285715,
      "paddingVertical": 21.57941176470588,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 1.5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "roboto_regular",
        "fontSize": 28.285714285714285,
        "textAlign": "center",
      }
    }
  >
    EN
  </Text>
</View>
`;

exports[`LanguageButton "fr" language and "size=small" renders according to snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#fff",
      "borderBottomWidth": 5,
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "paddingHorizontal": 55.714285714285715,
      "paddingVertical": 21.57941176470588,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 1.5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "roboto_regular",
        "fontSize": 28.285714285714285,
        "textAlign": "center",
      }
    }
  >
    FR
  </Text>
</View>
`;

exports[`LanguageButton "it" language and "size=small" renders according to snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#fff",
      "borderBottomWidth": 5,
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "paddingHorizontal": 55.714285714285715,
      "paddingVertical": 21.57941176470588,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 1.5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "roboto_regular",
        "fontSize": 28.285714285714285,
        "textAlign": "center",
      }
    }
  >
    IT
  </Text>
</View>
`;

exports[`LanguageButton selected button with "de" language 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DC0018",
      "borderBottomWidth": 5,
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "paddingHorizontal": 96.42857142857143,
      "paddingVertical": 29.426470588235293,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 1.5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "roboto_medium",
        "fontSize": 69.14285714285714,
        "textAlign": "center",
      }
    }
  >
    DE
  </Text>
</View>
`;

exports[`LanguageButton selected button with "en" language 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DC0018",
      "borderBottomWidth": 5,
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "paddingHorizontal": 96.42857142857143,
      "paddingVertical": 29.426470588235293,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 1.5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "roboto_medium",
        "fontSize": 69.14285714285714,
        "textAlign": "center",
      }
    }
  >
    EN
  </Text>
</View>
`;

exports[`LanguageButton selected button with "fr" language 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DC0018",
      "borderBottomWidth": 5,
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "paddingHorizontal": 96.42857142857143,
      "paddingVertical": 29.426470588235293,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 1.5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "roboto_medium",
        "fontSize": 69.14285714285714,
        "textAlign": "center",
      }
    }
  >
    FR
  </Text>
</View>
`;

exports[`LanguageButton selected button with "it" language 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#DC0018",
      "borderBottomWidth": 5,
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "paddingHorizontal": 96.42857142857143,
      "paddingVertical": 29.426470588235293,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 1.5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "roboto_medium",
        "fontSize": 69.14285714285714,
        "textAlign": "center",
      }
    }
  >
    IT
  </Text>
</View>
`;
